Added upload service

Omermigdal opened the pull request:
Same here regarding Swagger UI .. not allowed

Omermigdal committed:
Added upload service

Omermigdal committed:
modified upload - removed presigned URL

tal-alon commented on source/FilesServer/app/service.py:
```diff
@@ -15,3 +16,11 @@ async def s3_download(file_name: str) -> str:
+    s3_client = get_s3_client()
+    file_name = file.filename
+    s3_client.upload_fileobj(file.file, settings.aws_bucket_name, file_name)
+    return
```
remove redundant return statement

tal-alon commented on source/FilesServer/app/service.py:
```diff
@@ -15,3 +16,11 @@ async def s3_download(file_name: str) -> str:
     return presigned_url
+
+
+async def s3_upload(file: UploadFile = File(...)) -> None:
```
i think we can simplify the type annotation to:
...(file: File) -> ....

Omermigdal replied:
fixed

tal-alon commented on source/FilesServer/app/endpoints.py:
```diff
@@ -15,6 +15,10 @@ async def download(file_name: str) -> RedirectResponse:
-    return await service.s3_upload(file)
+@router.post("/upload", status_code=status.HTTP_200_OK)
+async def upload(file: UploadFile) -> None:
+    if not file:
```
has you checked if fastapi+pydantic are checking this for us?
file is annotated as "UploadFile" not "UploadFile | None" so i think it shouldn't be None-able. if so, this if statement is redundant

Omermigdal replied:
yup

Omermigdal committed:
- changed s3_download to s3_get_presigned_url
- added file not found handling
- changed upload status code to 201
- fixed tal comments

Omermigdal committed:
- changed s3_client = get_client to FastAPI depends

Omermigdal committed:
added is_file _exist and corrected accordingly
